azurerm/internal/services/azurestackhci/parse/hci_cluster.go

```diff
@@ -0,0 +1,45 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type HCIClusterId struct {
+	ResourceGroup string
+	Name          string
+}
+
+func (id HCIClusterId) ID(subscriptionId string) string {
+	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.AzureStackHCI/clusters/%s"
+	return fmt.Sprintf(fmtString, subscriptionId, id.ResourceGroup, id.Name)
+}
```

Comment:
We need to include the subscriptionId into the struct and do not use the subscriptionId parameter (which is being deprecated)
```suggestion
type HCIClusterId struct {
SubscriptionID string
	ResourceGroup string
	Name          string
}

func (id HCIClusterId) ID(_string) string {
	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.AzureStackHCI/clusters/%s"
	return fmt.Sprintf(fmtString, id.SubscriptionID, id.ResourceGroup, id.Name)
}
```

Reply:
updated